### PR TITLE
WT-9368 Extend WT fault injection testing to support LazyFS. 

### PR DESCRIPTION
**Pull request by pmacko86**

### CODE REVIEW

File: test/utility/lazyfs.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    testutil_die(ENOENT, "LazyFS is not available on this platform.");
+#else
+    char subdir_arg[PATH_MAX];
+    int e, count;
+    pid_t p, parent_pid, pid;
+
+    /*
+     * Mount in a separate process that will be automatically killed if the parent unexpectedly
+     * exits. In this way, we never forget to unmount.
+     */
```
Comment (ddanderson) [inline test/utility/lazyfs.c:137]: This is a cool technique.

**Review by ddanderson — APPROVED** [review]
Looks good to me.  Added some suggestions to tidy up a little bit.

File: test/csuite/README [hunk 2]
```diff
@@ -10,3 +10,11 @@ crafted and put into the Evergreen configuration file. We have a utility program
...
 Evergreen configuration for new or missing csuite tests. The program checking
 has been bound into developer workflow through 'dist/s_evergreen'. 
+
+Some csuite tests that check WiredTiger's failure recovery have LazyFS variants,
+which use LazyFS (a FUSE file system) to emulate losing the contents of the buffer
+cache, such as to test the system behavior in face of sudden reboots or power
+failures. To use these tests, you need to have FUSE 3 installed together with the
+development libraries, have 'user_allow_other' enabled in /etc/fuse.conf, and you
+need to also enable LazyFS in cmake via the '-DENABLE_LAZYFS=1' command-line
+option. The tests are not currently available in Evergreen and must be run manually.
```
[the remark this answers is not in the document]
Comment (pmacko86) [reply test/csuite/README:20]: Yes, automating these tests with Evergreen has been separated out from this ticket into WT-10591.

File: test/csuite/random_abort/main.c [hunk 3]
```diff
@@ -31,7 +31,13 @@
 #include <sys/wait.h>
 #include <signal.h>
 
-static char home[1024]; /* Program working dir */
+#ifdef __linux__
+#include <sys/prctl.h>
+#include <sys/types.h>
+#include <libgen.h>
```
[the remark this answers is not in the document]
Comment (pmacko86) [reply test/csuite/random_abort/main.c]: It turns out none of these includes are required any more after my refactoring; I forgot to remove these—thanks for spotting that!

File: test/csuite/schema_abort/main.c [hunk 4]
```diff
@@ -1089,8 +1108,51 @@ main(int argc, char *argv[])
...
+            testutil_check(__wt_snprintf(
+              lazyfs_logfile, sizeof(lazyfs_logfile), "%s/%s", home_canonical, LAZYFS_LOG_FILE));
+            lazyfs_create_config(lazyfs_config, lazyfs_control, lazyfs_logfile);
+
+            /* Mount LazyFS. */
+            testutil_check(__wt_snprintf(
+              lazyfs_mountpoint, sizeof(lazyfs_mountpoint), "%s/%s", home_canonical, WT_HOME_DIR));
+            lazyfs_pid = lazyfs_mount(lazyfs_mountpoint, lazyfs_base, lazyfs_config);
+        }
+
```
[the remark this answers is not in the document]
Comment (pmacko86) [reply test/csuite/schema_abort/main.c]: I just did that refactoring—thanks!

**Review by keitharnoldsmith — APPROVED** [review]
LGTM

Nice work!

**Pull request merged**